### ISSUE TITLE
PiVPN and Pihole | Connection times out intermittently

### ISSUE THREAD

**Issue by inphra** [pivpn/pivpn#646]
Hi all,

I have a random timing out issue with PiVPN that's making it a bit hard to google as it's conflating several unrelated issues.

The installation for Pihole and PiVPN went perfectly well with no issue;  I'm able to connect to PiVPN and Pihole worked perfectly fine.

However, once connected to PiVPN, at intermittent moments (possibly when network switching), I essentially lose connection (while still remaining connected to the VPN);  no apps work, can't ping any sites;  DNS essentially gets a bad probe / timeout error.

What resolves the issue are two things:
- Switching from WiFi to Network or vice versa and waiting 5-10 minutes
- Turn VPN off and on immediately resolves the issue

The key params I've changed in server.conf are:
- DIsabled / commented out:   push "redirect-gateway def1" so I only route DNS traffic
- Added "route 192.168.50.0 255.255.255.0"

Has anyone else encountered this issue before?

Thank you.

**Comment by inphra** [comment]
Anyone have any ideas why PiVPN would randomly stop routing DNS traffic and when reconnecting to the VPN or getting off the network would fix it?

**Comment by redstarksten** [comment]
Try to edit /etc/dnsmasq.conf

Add

listen-address=127.0.0.1
listen-address=10.8.0.1

Then, in /etc/openvpn/server.conf

Add push "dhcp-option DNS 10.8.0.1"

Delete any other lines about "dhcp-option"

service dnsmasq restart
service openvpn restart

**Comment by inphra** [comment]
> Try to edit /etc/dnsmasq.conf
> 
> Add
> 
> listen-address=127.0.0.1
> listen-address=10.8.0.1
> 
> Then, in /etc/openvpn/server.conf
> 
> Add push "dhcp-option DNS 10.8.0.1"
> 
> Delete any other lines about "dhcp-option"
> 
> service dnsmasq restart
> service openvpn restart
> 
> For detail you can look out from this link https://github.com/BobNisco/adblocking-vpn/blob/master/README.md
> 
> I am not combine pivpn + pihole anymore since too many conflict. If you just want make adblock run in vpn client connect, you can follow step in link above. For me, it is more simple, the function totally same as you edit file /etc/hosts.

Thank you so much for your suggestion.


I tried exactly what you suggested, but it broke the internet entirely;  sites were timing out; nothing was loading.  I undid the changes and it started working again.

Any ideas and why that might have happened?

Thanks again

**Comment by redstarksten** [comment]
> > Try to edit /etc/dnsmasq.conf
> > Add
> > listen-address=127.0.0.1
> > listen-address=10.8.0.1
> > Then, in /etc/openvpn/server.conf
> > Add push "dhcp-option DNS 10.8.0.1"
> > Delete any other lines about "dhcp-option"
> > service dnsmasq restart
> > service openvpn restart
> > For detail you can look out from this link https://github.com/BobNisco/adblocking-vpn/blob/master/README.md
> > I am not combine pivpn + pihole anymore since too many conflict. If you just want make adblock run in vpn client connect, you can follow step in link above. For me, it is more simple, the function totally same as you edit file /etc/hosts.
> 
> Thank you so much for your suggestion.
> 
> I tried exactly what you suggested, but it broke the internet entirely; sites were timing out; nothing was loading. I undid the changes and it started working again.
> 
> Any ideas and why that might have happened?
> 
> Thanks again

Are you keep using pihole + pivpn?
Like I said, I cant solve pivpn run both with pihole.
So, I try another way to block ads in client vpn. I just give an alternative, not to fix pihole + pivpn.

If you want to keep pivpn + pihole run together, try visit https://github.com/pi-hole/pi-hole/issues and describe your problem. Maybe someone out there can fix your issue.

**Comment by inphra** [comment]
> Pipvn and pihole run just fine, usually you only need to change the `push "dhcp-option DNS something"` to `push "dhcp-option DNS 10.8.0.1"` and allow local devices ("at most one hop away" as presented by the web interface).
> 
> > However, once connected to PiVPN, at intermittent moments (possibly when network switching), I essentially lose connection (while still remaining connected to the VPN); no apps work, can't ping any sites; DNS essentially gets a bad probe / timeout error.
> 
> I don't think this depends on the fact that you only route dns and local devices through the vpn, maybe your client is just losing connection to the server, thus dns times out.
> 
> Perhaps you could post the raspberry pi log `cat /var/log/openvpn.log`, remember to censor the client ip that will show up. Do you experience this behaviour on all clients? On what networks? Can you post the client log too?

Thanks for your suggestion.

I updated my config to `push "dhcp-option DNS 10.8.0.1"` instead of pointing to my Pi's local IP, however I'm still running into the issue.

I have two VPN clients (phones) that connect to my VPN;  it occurs on both, one of which is a brand new Pixel 3.  As soon as I pause or disconnect the VPN and resume/reconnect, all is well.  Happens on all networks.

I'll need to get the openvpn.log file for you though.

![pivpn_issue](https://user-images.githubusercontent.com/45575595/49835295-9d6bca00-fd53-11e8-9c56-83c44a782640.gif)

**Comment by inphra** [comment]
> 
> 
> Pipvn and pihole run just fine, usually you only need to change the `push "dhcp-option DNS something"` to `push "dhcp-option DNS 10.8.0.1"` and allow local devices ("at most one hop away" as presented by the web interface).
> 
> > However, once connected to PiVPN, at intermittent moments (possibly when network switching), I essentially lose connection (while still remaining connected to the VPN); no apps work, can't ping any sites; DNS essentially gets a bad probe / timeout error.
> 
> I don't think this depends on the fact that you only route dns and local devices through the vpn, maybe your client is just losing connection to the server, thus dns times out.
> 
> Perhaps you could post the raspberry pi log `cat /var/log/openvpn.log`, remember to censor the client ip that will show up. Do you experience this behaviour on all clients? On what networks? Can you post the client log too?

Hey friend,

Apologies for the late response.

I've attached my openvpn.log.   Hopefully this reveals the problem.

Thank you!


Dec 16 14:00:16 pi ovpn-server[462]: TLS Error: cannot locate HMAC in incoming packet from [AF_INET][External IP]:46998
Dec 16 14:45:50 pi ovpn-server[462]: [External IP]:10031 TLS: Initial packet from [AF_INET][External IP]:10031, sid=2bd8370f af2d10c8
Dec 16 14:45:50 pi ovpn-server[462]: [External IP]:10031 VERIFY OK: depth=1, CN=ChangeMe
Dec 16 14:45:50 pi ovpn-server[462]: [External IP]:10031 Validating certificate key usage
Dec 16 14:45:50 pi ovpn-server[462]: [External IP]:10031 ++ Certificate has key usage  0080, expects 0080
Dec 16 14:45:50 pi ovpn-server[462]: [External IP]:10031 VERIFY KU OK
Dec 16 14:45:50 pi ovpn-server[462]: [External IP]:10031 Validating certificate extended key usage
Dec 16 14:45:50 pi ovpn-server[462]: [External IP]:10031 ++ Certificate has EKU (str) TLS Web Client Authentication, expects TLS Web Client Authentication
Dec 16 14:45:50 pi ovpn-server[462]: [External IP]:10031 VERIFY EKU OK
Dec 16 14:45:50 pi ovpn-server[462]: [External IP]:10031 VERIFY OK: depth=0, CN=my-dns
Dec 16 14:45:51 pi ovpn-server[462]: [External IP]:10031 peer info: IV_GUI_VER=OC30Android
Dec 16 14:45:51 pi ovpn-server[462]: [External IP]:10031 peer info: IV_VER=3.2
Dec 16 14:45:51 pi ovpn-server[462]: [External IP]:10031 peer info: IV_PLAT=android
Dec 16 14:45:51 pi ovpn-server[462]: [External IP]:10031 peer info: IV_NCP=2
Dec 16 14:45:51 pi ovpn-server[462]: [External IP]:10031 peer info: IV_TCPNL=1
Dec 16 14:45:51 pi ovpn-server[462]: [External IP]:10031 peer info: IV_PROTO=2
Dec 16 14:45:51 pi ovpn-server[462]: [External IP]:10031 peer info: IV_AUTO_SESS=1
Dec 16 14:45:51 pi ovpn-server[462]: [External IP]:10031 Control Channel: TLSv1.2, cipher TLSv1/SSLv3 D8AEF-RSA-AES256-GCM-SHA384, 2048 bit RSA
Dec 16 14:45:51 pi ovpn-server[462]: [External IP]:10031 [my-dns] Peer Connection Initiated with [AF_INET][External IP]:10031
Dec 16 14:45:51 pi ovpn-server[462]: my-dns/[External IP]:10031 MULTI_sva: pool returned IPv4=10.8.0.2, IPv6=(Not enabled)
Dec 16 14:45:51 pi ovpn-server[462]: my-dns/[External IP]:10031 MULTI: Learn: 10.8.0.2 -> my-dns/[External IP]:10031
Dec 16 14:45:51 pi ovpn-server[462]: my-dns/[External IP]:10031 MULTI: primary virtual IP for my-dns/[External IP]:10031: 10.8.0.2
Dec 16 14:45:51 pi ovpn-server[462]: my-dns/[External IP]:10031 PUSH: Received control message: 'PUSH_REQUEST'
Dec 16 14:45:51 pi ovpn-server[462]: my-dns/[External IP]:10031 SENT CONTROL [my-dns]: 'PUSH_REPLY,dhcp-option DNS 10.8.0.1,block-outside-dns,route 192.168.50.0 255.255.255.0,route-gateway 10.8.0.1,topology subnet,ping 1800,ping-restart 3600,ifconfig 10.8.0.2 255.255.255.0,peer-id 0,cipher AES-256-GCM' (status=1)
Dec 16 14:45:51 pi ovpn-server[462]: my-dns/[External IP]:10031 Data Channel Encrypt: Cipher 'AES-256-GCM' initialized with 256 bit key
Dec 16 14:45:51 pi ovpn-server[462]: my-dns/[External IP]:10031 Data Channel Decrypt: Cipher 'AES-256-GCM' initialized with 256 bit key
Dec 16 15:45:51 pi ovpn-server[462]: my-dns/[External IP]:46668 TLS: soft reset sec=0 bytes=24906/-1 pkts=231/0
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 VERIFY OK: depth=1, CN=ChangeMe
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 Validating certificate key usage
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 ++ Certificate has key usage  0080, expects 0080
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 VERIFY KU OK
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 Validating certificate extended key usage
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 ++ Certificate has EKU (str) TLS Web Client Authentication, expects TLS Web Client Authentication
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 VERIFY EKU OK
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 VERIFY OK: depth=0, CN=my-dns
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 peer info: IV_GUI_VER=OC30Android
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 peer info: IV_VER=3.2
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 peer info: IV_PLAT=android
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 peer info: IV_NCP=2
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 peer info: IV_TCPNL=1
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 peer info: IV_PROTO=2
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 peer info: IV_AUTO_SESS=1
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1569', remote='link-mtu 1524'
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 WARNING: 'cipher' is used inconsistently, local='cipher AES-256-CBC', remote='cipher AES-256-GCM'
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 Data Channel Encrypt: Cipher 'AES-256-GCM' initialized with 256 bit key
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 Data Channel Decrypt: Cipher 'AES-256-GCM' initialized with 256 bit key
Dec 16 15:46:07 pi ovpn-server[462]: my-dns/[External IP]:2422 Control Channel: TLSv1.2, cipher TLSv1/SSLv3 D8AEF-RSA-AES256-GCM-SHA384, 2048 bit RSA
Dec 16 16:15:28 pi ovpn-server[462]: my-dns/[External IP]:25627 SIGTERM[soft,remote-exit] received, client-instance exiting
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 TLS: Initial packet from [AF_INET][External IP]:47248, sid=a1807b07 7d656dc5
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 Authenticate/Decrypt packet error: bad packet ID (may be a replay): [ #1 / time = (1545005766) Sun Dec 16 16:16:06 2018 ] -- see the man page entry for --no-replay and --replay-window for more info or silence this warning with --mute-replay-warnings
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 TLS Error: incoming packet authentication failed from [AF_INET][External IP]:47248
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 VERIFY OK: depth=1, CN=ChangeMe
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 Validating certificate key usage
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 ++ Certificate has key usage  0080, expects 0080
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 VERIFY KU OK
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 Validating certificate extended key usage
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 ++ Certificate has EKU (str) TLS Web Client Authentication, expects TLS Web Client Authentication
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 VERIFY EKU OK
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 VERIFY OK: depth=0, CN=my-dns
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 peer info: IV_GUI_VER=OC30Android
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 peer info: IV_VER=3.2
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 peer info: IV_PLAT=android
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 peer info: IV_NCP=2
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 peer info: IV_TCPNL=1
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 peer info: IV_PROTO=2
Dec 16 16:16:09 pi ovpn-server[462]: [External IP]:47248 peer info: IV_AUTO_SESS=1
Dec 16 16:16:10 pi ovpn-server[462]: [External IP]:47248 Control Channel: TLSv1.2, cipher TLSv1/SSLv3 D8AEF-RSA-AES256-GCM-SHA384, 2048 bit RSA
Dec 16 16:16:10 pi ovpn-server[462]: [External IP]:47248 [my-dns] Peer Connection Initiated with [AF_INET][External IP]:47248
Dec 16 16:16:10 pi ovpn-server[462]: my-dns/[External IP]:47248 MULTI_sva: pool returned IPv4=10.8.0.2, IPv6=(Not enabled)
Dec 16 16:16:10 pi ovpn-server[462]: my-dns/[External IP]:47248 MULTI: Learn: 10.8.0.2 -> my-dns/[External IP]:47248
Dec 16 16:16:10 pi ovpn-server[462]: my-dns/[External IP]:47248 MULTI: primary virtual IP for my-dns/[External IP]:47248: 10.8.0.2
Dec 16 16:16:10 pi ovpn-server[462]: my-dns/[External IP]:47248 PUSH: Received control message: 'PUSH_REQUEST'
Dec 16 16:16:10 pi ovpn-server[462]: my-dns/[External IP]:47248 SENT CONTROL [my-dns]: 'PUSH_REPLY,dhcp-option DNS 10.8.0.1,block-outside-dns,route 192.168.50.0 255.255.255.0,route-gateway 10.8.0.1,topology subnet,ping 1800,ping-restart 3600,ifconfig 10.8.0.2 255.255.255.0,peer-id 0,cipher AES-256-GCM' (status=1)
Dec 16 16:16:10 pi ovpn-server[462]: my-dns/[External IP]:47248 Data Channel Encrypt: Cipher 'AES-256-GCM' initialized with 256 bit key
Dec 16 16:16:10 pi ovpn-server[462]: my-dns/[External IP]:47248 Data Channel Decrypt: Cipher 'AES-256-GCM' initialized with 256 bit key
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 TLS: Initial packet from [AF_INET][External IP]:58797, sid=afa1e1e4 22389fa2
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 VERIFY OK: depth=1, CN=ChangeMe
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 Validating certificate key usage
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 ++ Certificate has key usage  0080, expects 0080
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 VERIFY KU OK
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 Validating certificate extended key usage
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 ++ Certificate has EKU (str) TLS Web Client Authentication, expects TLS Web Client Authentication
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 VERIFY EKU OK
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 VERIFY OK: depth=0, CN=my-dns
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 peer info: IV_GUI_VER=OC30Android
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 peer info: IV_VER=3.2
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 peer info: IV_PLAT=android
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 peer info: IV_NCP=2
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 peer info: IV_TCPNL=1
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 peer info: IV_PROTO=2
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 peer info: IV_AUTO_SESS=1
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 Control Channel: TLSv1.2, cipher TLSv1/SSLv3 D8AEF-RSA-AES256-GCM-SHA384, 2048 bit RSA
Dec 16 17:02:17 pi ovpn-server[462]: [External IP]:58797 [my-dns] Peer Connection Initiated with [AF_INET][External IP]:58797
Dec 16 17:02:17 pi ovpn-server[462]: MULTI: new connection by client 'my-dns' will cause previous active sessions by this client to be dropped.  Remember to use the --duplicate-cn option if you want multiple clients using the same certificate or username to concurrently connect.
Dec 16 17:02:17 pi ovpn-server[462]: MULTI_sva: pool returned IPv4=10.8.0.2, IPv6=(Not enabled)
Dec 16 17:02:17 pi ovpn-server[462]: MULTI: Learn: 10.8.0.2 -> my-dns/[External IP]:58797
Dec 16 17:02:17 pi ovpn-server[462]: MULTI: primary virtual IP for my-dns/[External IP]:58797: 10.8.0.2
Dec 16 17:02:17 pi ovpn-server[462]: my-dns/[External IP]:58797 PUSH: Received control message: 'PUSH_REQUEST'
Dec 16 17:02:17 pi ovpn-server[462]: my-dns/[External IP]:58797 SENT CONTROL [my-dns]: 'PUSH_REPLY,dhcp-option DNS 10.8.0.1,block-outside-dns,route 192.168.50.0 255.255.255.0,route-gateway 10.8.0.1,topology subnet,ping 1800,ping-restart 3600,ifconfig 10.8.0.2 255.255.255.0,peer-id 1,cipher AES-256-GCM' (status=1)
Dec 16 17:02:17 pi ovpn-server[462]: my-dns/[External IP]:58797 Data Channel Encrypt: Cipher 'AES-256-GCM' initialized with 256 bit key
Dec 16 17:02:17 pi ovpn-server[462]: my-dns/[External IP]:58797 Data Channel Decrypt: Cipher 'AES-256-GCM' initialized with 256 bit key
Dec 16 17:56:55 pi ovpn-server[462]: [External IP]:64177 TLS: Initial packet from [AF_INET][External IP]:64177, sid=6cf568a7 b8c554ff
Dec 16 17:56:55 pi ovpn-server[462]: [External IP]:64177 Authenticate/Decrypt packet error: bad packet ID (may be a replay): [ #1 / time = (1545011810) Sun Dec 16 17:56:50 2018 ] -- see the man page entry for --no-replay and --replay-window for more info or silence this warning with --mute-replay-warnings
Dec 16 17:56:55 pi ovpn-server[462]: [External IP]:64177 TLS Error: incoming packet authentication failed from [AF_INET][External IP]:64177
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 VERIFY OK: depth=1, CN=ChangeMe
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 Validating certificate key usage
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 ++ Certificate has key usage  0080, expects 0080
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 VERIFY KU OK
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 Validating certificate extended key usage
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 ++ Certificate has EKU (str) TLS Web Client Authentication, expects TLS Web Client Authentication
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 VERIFY EKU OK
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 VERIFY OK: depth=0, CN=my-dns
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 peer info: IV_GUI_VER=OC30Android
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 peer info: IV_VER=3.2
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 peer info: IV_PLAT=android
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 peer info: IV_NCP=2
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 peer info: IV_TCPNL=1
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 peer info: IV_PROTO=2
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 peer info: IV_AUTO_SESS=1
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 Control Channel: TLSv1.2, cipher TLSv1/SSLv3 D8AEF-RSA-AES256-GCM-SHA384, 2048 bit RSA
Dec 16 17:56:56 pi ovpn-server[462]: [External IP]:64177 [my-dns] Peer Connection Initiated with [AF_INET][External IP]:64177
Dec 16 17:56:56 pi ovpn-server[462]: MULTI: new connection by client 'my-dns' will cause previous active sessions by this client to be dropped.  Remember to use the --duplicate-cn option if you want multiple clients using the same certificate or username to concurrently connect.
Dec 16 17:56:56 pi ovpn-server[462]: MULTI_sva: pool returned IPv4=10.8.0.2, IPv6=(Not enabled)
Dec 16 17:56:56 pi ovpn-server[462]: MULTI: Learn: 10.8.0.2 -> my-dns/[External IP]:64177
Dec 16 17:56:56 pi ovpn-server[462]: MULTI: primary virtual IP for my-dns/[External IP]:64177: 10.8.0.2
Dec 16 17:56:56 pi ovpn-server[462]: my-dns/[External IP]:64177 PUSH: Received control message: 'PUSH_REQUEST'
Dec 16 17:56:56 pi ovpn-server[462]: my-dns/[External IP]:64177 SENT CONTROL [my-dns]: 'PUSH_REPLY,dhcp-option DNS 10.8.0.1,block-outside-dns,route 192.168.50.0 255.255.255.0,route-gateway 10.8.0.1,topology subnet,ping 1800,ping-restart 3600,ifconfig 10.8.0.2 255.255.255.0,peer-id 0,cipher AES-256-GCM' (status=1)
Dec 16 17:56:56 pi ovpn-server[462]: my-dns/[External IP]:64177 Data Channel Encrypt: Cipher 'AES-256-GCM' initialized with 256 bit key
Dec 16 17:56:56 pi ovpn-server[462]: my-dns/[External IP]:64177 Data Channel Decrypt: Cipher 'AES-256-GCM' initialized with 256 bit key

**Comment by Trotter73** [comment]
Hi,

A me too comment I'm afraid, I can add some more flesh to it though.

I am not running PiHole, so maybe that eliminates some variables...

With regards to clients, I am connecting via an Android phone, the timeout almost exclusively happens when switching from a network connection to WiFi and vice versa. I have had the phone on WiFi all day with no issues, similarly with a network connection, switching seems to cause the problem.

I have found that OpenVPN for Android (0.7.5) by Arne Schwabe is less susceptible to this than the official OpenVPN Connect client, it still does happen though.

I think, the issue started when I updated to V1.10 earlier in the year, I cant be 100% sure of this though..

I have not tried the fixes above as they do not appear to have worked, I can do if need be though.

Let me know if / how I can help...

**Comment by orazioedoardo** [comment]
Can you post these files? `cat /etc/openvpn/server.conf` and 'cat /etc/openvpn/easy-rsa/pki/Default.txt'. In the OpenVPN Connect app do you have compression enabled in settings?

**Comment by Trotter73** [comment]
Hi,

Sorry for the late reply..

[new 1.txt](https://github.com/pivpn/pivpn/files/2720438/new.1.txt)
[new 2.txt](https://github.com/pivpn/pivpn/files/2720439/new.2.txt)



The app is not showing LZO compression as being enabled...

**Comment by inphra** [comment]
> Can you post these files? `cat /etc/openvpn/server.conf` and `cat /etc/openvpn/easy-rsa/pki/Default.txt` (you may want to censor the IP in the second one). In the OpenVPN Connect app do you have compression enabled in settings?

Hi friend,  apologies for the delay -- the holiday season took its toll!


As you requested, here are the two files:

[Default.txt](https://github.com/pivpn/pivpn/files/2735364/Default.txt)
[server.txt](https://github.com/pivpn/pivpn/files/2735365/server.txt)

In the OVPN client app for Android, compression is set to "Full"

**Comment by orazioedoardo** [comment]
@inphra In the openvpn client app disable compression.

@Trotter73 Edit both the server config and client template with `sudo nano /etc/openvpn/server.conf` and `sudo nano /etc/openvpn/easy-rsa/pki/Default.txt` and delete the line `compress lz4`.
Save and restart the server with `sudo systemctl restart openvpn`.
Remove the same line from your .ovpn files, then try to connect.
Make sure that you have compression (insecure) set to no in the app.

**Comment by Trotter73** [comment]
Hi,

Cool, I've made the edits so will see how we get on, I'll report back in a day or so after some proper running.. Thanks for your help, appreciated !

**Comment by inphra** [comment]
> @inphra In the openvpn client app disable compression (set to "No").


Update - @orazioedoardo, unfortunately disabling compression did not resolve the issue.  I'm consistently running into the issue when I hop between networks.

Any thoughts on what else I might look at?

Thank you for your help.

**Comment by inphra** [comment]
> @inphra have you tried OpenVPN for Android (0.7.5) by Arne Schwabe ? I have found this to be better than the official client...

I've tried it as per your suggestion.  I'm definitely still encountering the same issue, but I can't comment yet which is worse, OpenVPN original client versus Arne Schwabe's client.

It's definitely associated with network hopping.

**Comment by orazioedoardo** [comment]
Maybe you can change the `keepalive 1800 3600` to something shorter such that the peers will ping each other more often (every 10 seconds) and restart if unsuccessful (after 120 second): `keepalive 10 120`.

**Comment by Trotter73** [comment]
In light of the abandonment notice I feel happier posting this, I found this script that works perfectly on the Pi and appears to be maintained, I have been running it for a week or so now with no issues, including the dreaded timeout reported in this thread.. 
[https://github.com/angristan/openvpn-install](url)

**Comment by Firemanjoe** [comment]
> Maybe you can change the `keepalive 1800 3600` to something shorter such that the peers will ping each other more often (every 10 seconds) and restart if unsuccessful (after 120 second): `keepalive 10 120`.
> https://openvpn.net/community-resources/reference-manual-for-openvpn-2-4/

I'm trying to do this and it isn't working for me either. Does anyone have any suggestions or is there progress on this one?

**Comment by 4s3ti** [comment]
is this still a thing? can this be closed? this has been without any activity for a while.

**Comment by inphra** [comment]
It's still happening unfortunately.

**Comment by 4s3ti** [comment]
Ok thanks, 
I will keep this open in case anyone knows how to help you out... 
if i have time in the meanwhile i will try to pay a bit more attention to this and see if there is anything i can do to help out. 

Kind Regards.

**Comment by inphra** [comment]
Really appreciate it.  It wasn't a problem before. Thinking of rewiping and
starting my pi build from scratch.

**Comment by inphra** [comment]
Finally got around to rebuilding my Pihole + pivpn set up last night.

Apart from setting it up it normally, I only made one modification and that was to tunnel DNS traffic only versus all traffic. 

With the openvpn client, I've also made sure that battery saving optimizations are not turned on on my Android device.  I have a feeling that this may be the culprit but I have yet to test this theory; was battered optimization set to on for the app itself, it will drop connection or so I've read.  I'll report back once I know for sure!
